kvgraph/interface.go

```diff
@@ -0,0 +1,61 @@
+package kvgraph
+
+import (
+	"fmt"
+	"github.com/bmeg/arachne/gdbi"
+)
+
+type KVBuilder func(path string) (KVInterface, error)
+
+type KVInterface interface {
+	HasKey(key []byte) bool
+	Set(key, value []byte) error
+	DeletePrefix(prefix []byte) error
+	Delete(key []byte) error
+
+	View(func(it KVIterator) error) error
+	Update(func(tx KVTransaction) error) error
+	Close() error
+}
+
+type KVIterator interface {
+	Seek(k []byte) error
+	Valid() bool
+	Key() []byte
+	Value() ([]byte, error)
+	Next() error
+
+	Get(key []byte) ([]byte, error)
```

Comment:
Are you sure you want your primary interface to take `[]byte`? Will this always be a string?

Reply:
Most of the time its '0x00' joined strings.
For example, the Vertex key pattern is
'v' 0x00 $vertex_id 0x00 $label

The src edge index pattern is 
's' 0x00 $src_id 0x00 $dst_id 0x00 $edge_id 0x00 $label

Reply:
If there are a small, limited set of key patterns, it's probably better to either make those separate methods in the interface, or to provide key helpers (a la Datastore's [NameKey](https://godoc.org/cloud.google.com/go/datastore#NameKey)).